tests/go/go-kafka-serde/client.go

```diff
@@ -267,13 +253,110 @@ func (tc *TestClient) RunTest() error {
 			}
 
 			numConsumed++
-		case kafka.Error:
-			util.Die("Failure during consumption: %v", e)
+		case *kafka.Error:
+			return e
 		default:
 			log.Debugf("Ignoring unknown message: %v", e)
 		}
 	}
 
+	log.Debugf("Consumed %v messages", numConsumed)
+
+	if numConsumed != tc.count {
+		util.Die("Did not consume expected number of messages: %v != %v", numConsumed, tc.count)
+	}
+
+	return nil
+}
+
+func (tc *TestClient) produceMessages() error {
+	numDelivered := 0
+
+	// Create a context with a cancel function
+	produceCheckContext, cancelProduceCheck := context.WithCancel(context.Background())
+	// Create an error group that will be used to monitor for cancellation and
+	// message production events from franz-go producer
+	produceCheckGroup, produceCheckContext := errgroup.WithContext(produceCheckContext)
+	// Create a timeout context
```

Comment:
nit: But, why?

Reply:
why what?

Reply:
I think he's saying the comment is saying what is happening which *should* be self explanatory from the code. Comments are more useful if they are say why something is happening, otherwise comments like this are likely to become stale.

Reply:
Yes. Sorry, I should have used my words.